File: medical_organizer/lib/models/allergy.dart
class Allergy {
  final int id;
  final int? relativeId;
  final String? relativeName;
  final String? relationType;
  final String allergen;
  final String? reaction;
  final String? severity;

  Allergy({
    required this.id,
    this.relativeId,
    this.relativeName,
    this.relationType,
    required this.allergen,
    this.reaction,
    this.severity,
  });

  factory Allergy.fromJson(Map<String, dynamic> j) => Allergy(
        id: j['id'],
        relativeId: j['relative_id'],
        relativeName: j['relative_name'],
        relationType: j['relation_type'],
        allergen: j['allergen'],
        reaction: j['reaction'],
        severity: j['severity'],
      );

  Map<String, dynamic> toJson() => {
        'relative_id': relativeId,
        'allergen': allergen,
        'reaction': reaction,
        'severity': severity,
      };

  static String severityLabel(String? s) {
    const map = {'mild': 'Лёгкая', 'moderate': 'Средняя', 'severe': 'Тяжёлая'};
    return map[s] ?? '—';
  }
}

File: server/models/Allergy.js
const pool = require('../db/database');

const Allergy = {
  async findAll(userId) {
    const { rows } = await pool.query(
      `SELECT a.*, r.full_name AS relative_name, r.relation_type
       FROM allergies a
       LEFT JOIN relatives r ON a.relative_id = r.id
       WHERE a.user_id=$1
       ORDER BY a.created_at DESC`,
      [userId]
    );
    return rows;
  },

  async findOne(id, userId) {
    const { rows } = await pool.query(
      'SELECT * FROM allergies WHERE id=$1 AND user_id=$2', [id, userId]
    );
    return rows[0] || null;
  },

  async create({ user_id, relative_id, allergen, reaction, severity }) {
    const { rows } = await pool.query(
      `INSERT INTO allergies (user_id, relative_id, allergen, reaction, severity)
       VALUES ($1,$2,$3,$4,$5) RETURNING id`,
      [user_id, relative_id || null, allergen, reaction || null, severity || null]
    );
    return rows[0].id;
  },

  async update(id, userId, { relative_id, allergen, reaction, severity }) {
    const { rowCount } = await pool.query(
      `UPDATE allergies SET relative_id=$1, allergen=$2, reaction=$3, severity=$4
       WHERE id=$5 AND user_id=$6`,
      [relative_id || null, allergen, reaction || null, severity || null, id, userId]
    );
    return { changes: rowCount };
  },

  async delete(id, userId) {
    const { rowCount } = await pool.query(
      'DELETE FROM allergies WHERE id=$1 AND user_id=$2', [id, userId]
    );
    return { changes: rowCount };
  },
};

module.exports = Allergy;
